[3/3] Refactors lib/private/Security

fsamapoor opened the pull request:
Following [previous PRs taking advantage of PHP8's constructor property promotion](https://github.com/nextcloud/server/pulls?q=is%3Apr+author%3Afsamapoor+Uses+PHP8%27s+constructor+property+promotion) in `/core/` namespace, I have also made the required adjustments to the classes in `/lib/private/Security` namespace.

I figured I should split the changes into three PRs to make reviewing the changes easier.

The improvements in this PR include but are not limited to:

- Using PHP8's constructor property promotion
- Adding return types
- Adding types to properties
- Removing redundant docblocks

fsamapoor committed:
Refactors lib/private/Security.

Mainly using PHP8's constructor property promotion.

Signed-off-by: Faraz Samapoor <fsa@adlas.at>

come-nc commented on lib/private/Security/Certificate.php:
```diff
@@ -30,25 +30,24 @@
+	protected string $name;
 
-	protected $commonName;
+	protected mixed $commonName;
```
Why the `mixed` types? It seems to be `?string` since getCommonName returns `?string`.

fsamapoor replied:
It is being evaluated based on `$info['subject']['CN']` and I wasn't sure if its value would be of type `string`, but now that you mentioned it, I think it is safe to type it as it's getter's return type, which is `?string`.
I'll update other properties as well.

fsamapoor replied:
@come-nc One other thing. I couldn't figure out the property type for the `$serial` since my IDE didn't find any usage of that property in the project. How could I know if it is needed or it is redundant?

fsamapoor committed:
Updates the typed properties.

Based on: https://github.com/nextcloud/server/pull/39013#discussion_r1242340826

Co-authored-by: Côme Chilliet <91878298+come-nc@users.noreply.github.com>
Signed-off-by: Faraz Samapoor <fsa@adlas.at>

come-nc commented on lib/private/Security/Certificate.php:
```diff
@@ -30,25 +30,24 @@
-	protected $organization;
+	protected ?string $organization;
 
 	protected $serial;
```
```suggestion
```

Indeed it is unused, always has been, please remove it.

come-nc reviewed (approved):
Good otherwise

fsamapoor committed:
Update lib/private/Security/Certificate.php

Co-authored-by: Côme Chilliet <91878298+come-nc@users.noreply.github.com>
Signed-off-by: Faraz Samapoor <f.samapoor@gmail.com>

Fenn-CS commented:
Hello, @fsamapoor please rebase.

fsamapoor commented:
> Hello, @fsamapoor please rebase.

Messed it up a bit. I'll rebase it again to remove the merge commit and let you know afterward.

fsamapoor commented:
> Hello, @fsamapoor please rebase.

Rebased and removed the merge commit.
